subprojects/core/src/main/java/org/gradle/profile/FragmentedOperation.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.profile;
+
+import com.google.common.collect.Sets;
+
+import java.util.Collection;
+
+public class FragmentedOperation extends Operation {
+    private final String description;
+    private final Collection<ContinuousOperation> children = Sets.newConcurrentHashSet();
```

Comment:
Why does this need to be a `ConcurrentHashSet`? I thought our listener infrastructure made sure the listeners are served single threaded?

Reply:
Hm, I didn't know that. Where does it do it?

Reply:
See https://github.com/gradle/gradle/blob/0c4a1ab113654969fd2679e9e201b4ef75daad71/subprojects/messaging/src/main/java/org/gradle/internal/event/ListenerManager.java#L58.